src/meltano/cli/config.py

```diff
@@ -31,6 +32,62 @@
 logger = logging.getLogger(__name__)
 
 
+def _get_ctx_arg(*args):
+    """Get the click.core.Context arg from a set of args.
+
+    Args:
+        args: the args to get Context from
+
+    Returns:
+        The click.core.Context arg.
+
+    Raises:
+        ValueError: if there is no click.core.Context in the given args.
+    """
+    for arg in args:
+        if isinstance(arg, click.core.Context):
+            return arg
+    raise ValueError("No click.core.Context provided in *args")
+
+
+def _get_store_choices():
+    """Get a list of valid choices for the --store flag.
+
+    Returns:
+        SettingValueStore.writables(), without meltano_env
+    """
+    writables = SettingValueStore.writables()
+    writables.remove(SettingValueStore.MELTANO_ENV)
+    return writables
+
+
+def _use_meltano_env(func):
+    """Override the 'meltano_yml' choice for a config command's 'store' argument.
+
+    If an --environment flag is passed, the decorated command will use
+    the MELTANO_ENV store instead of MELTANO_YML but _will not_ use MELTANO_ENV
+    store if the active environment is set via the default environment.
+
+    Args:
+       func: the command to override
+
+    Returns:
+       A wrapped function with overridden store argument
+    """
+
+    @wraps(func)
+    def _wrapper(*args, **kwargs):
+        store = kwargs.pop("store")
+        if store not in {SettingValueStore.MELTANO_YML, SettingValueStore.MELTANO_ENV}:
+            return func(*args, **kwargs, store=store)
+        ctx = _get_ctx_arg(*args)
+        if ctx.obj["is_default_environment"]:
+            return func(*args, **kwargs, store=SettingValueStore.MELTANO_YML)
+        return func(*args, **kwargs, store=SettingValueStore.MELTANO_ENV)
```

Comment:
We can reduce the number of return statements here:

```suggestion
        store = (
            SettingValueStore.MELTANO_YML
            if ctx.obj["is_default_environment"]
            else SettingValueStore.MELTANO_ENV
        )
        return func(*args, **kwargs, store=store)
```